src/rewards/voters.rs

```diff
@@ -33,14 +37,65 @@ pub fn calculate_stake<'address>(
 pub fn calculate_reward_share<'address>(
     total_stake: u64,
     stake_per_voter: &HashMap<&'address Address, u64>,
+    threshold_addresses: &AddressesVoteCount,
+    threshold: u64,
 ) -> HashMap<&'address Address, U64F64> {
     stake_per_voter
         .iter()
-        .map(|(k, v)| (*k, fixed::types::U64F64::from_num(*v) / total_stake as u128))
+        .map(|(k, v)| {
+            let reward = if *threshold_addresses
+                .get(k)
+                .unwrap_or_else(|| panic!("Missing address {} in vote count map", k))
```

Comment:
Would this panic if a user was in block0 but did not actually vote?

Reply:
Not sure. It depends on chain implementation now that you mention it. Maybe we could consider a defaulting 0 votes a better choice though.